### PR TITLE
Added accessTier check to download call for BlobHandler.ts to fail on…

### PR DESCRIPTION
**Pull request by BrendonK**
… archived blobs.

#2473

Thanks for contribution! Please go through following checklist before sending PR.

### PR Branch Destination

- For Azurite V3, please send PR to `main` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Add Change Log

Add change log for the code change in `Upcoming Release` section in `ChangeLog.md`.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

**Comment by BrendonK** [comment]
@microsoft-github-policy-service agree

File: src/blob/handlers/BlobHandler.ts [hunk 1]
```diff
@@ -80,6 +80,10 @@ export default class BlobHandler extends BaseHandler implements IBlobHandler {
       options.modifiedAccessConditions
     );
 
+    if (blob.properties.accessTier === Models.AccessTier.Archive) {
+      throw StorageErrorFactory.getBlobArchived(context.contextId!);
```
Comment (blueww) [inline src/blob/handlers/BlobHandler.ts:84]: Please make sure the error returned here is same as the error return from product Azure server. (status code, error code, error message.)

Comment (BrendonK) [reply src/blob/handlers/BlobHandler.ts:84]: @blueww I got the error codes/message from [here](https://learn.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes) so it should be okay I think:
BlobArchived	Conflict (409)	This operation is not permitted on an archived blob.

Comment (BrendonK) [reply src/blob/handlers/BlobHandler.ts:84]: @blueww some tests are failing - I will check the failures to see what's causing it.

**Review by blueww — APPROVED** [review]

File: tests/blob/apis/blob.test.ts [hunk 2]
```diff
@@ -253,6 +253,22 @@ describe("BlobAPIs", () => {
     assert.fail();
   });
 
+  it("download should not work when blob in Archive tier", async () => {
```
Comment (blueww) [inline tests/blob/apis/blob.test.ts:256]: Please add `@loki @sql` as other test cases

File: tests/blob/sas.test.ts [hunk 3]
```diff
@@ -2122,7 +2122,7 @@ describe("Shared Access Signature (SAS) authentication", () => {
     }
     assert.ok(error !== undefined);
     assert.equal(error.statusCode, 409);
-    assert.equal(error.details.code, "BlobArchived");
+    assert.equal(error.details.code, "CannotVerifyCopySource");
```
Comment (blueww) [inline tests/blob/sas.test.ts:2125]: The SQL test case failure on this, looks related with `sourceBlob.setAccessTier("Archive");` on line 2113 don't have await, it's async call, without await the access tier might still not convert to archive.

Add await should can resolve the failure.
 `await sourceBlob.setAccessTier("Archive");`

Comment (blueww) [reply tests/blob/sas.test.ts:2125]: I have fixed the case

**Pull request merged**